src/browser/main.coffee

```diff
@@ -57,11 +57,14 @@ handleStartupEventWithSquirrel = ->
 setupCrashReporter = ->
   crashReporter.start(productName: 'Atom', companyName: 'GitHub')
 
-setupAtomHome = ->
+setupAtomHome = (args) ->
   return if process.env.ATOM_HOME
   atomHome = path.join(app.getHomeDir(), '.atom')
-  try
-    atomHome = fs.realpathSync(atomHome)
+  AtomPortable = require './atom-portable'
+  atomHome = AtomPortable.portableAtomHomePath() if AtomPortable.isPortableInstall process.platform, process.env.ATOM_HOME
+  atomHome = fs.realpathSync(atomHome)
```

Comment:
I think `fs.realpathSync` throws an error when it doesn't exist so I think we might still want to have a `try` statement around it and just ignore the error when it doesn't exist.